src/cv/saliency.cpp

```diff
@@ -1,3 +1,142 @@
 #include "cv/saliency.hpp"
+#include <torch/script.h> // One-stop header.
+#include <opencv2/opencv.hpp>
+#include <iostream>
+#include <memory>
+#include "cv/saliency.hpp"
+#include <torch/cuda.h>
+#include <torch/mps.h>
+#include "utilities/logging.hpp"
+
+// constructor definition
+Saliency::Saliency () {
+    this->modelPath = "";
+} 
+
+// constructor definition
+Saliency::Saliency (std::string modelPath) {
+    this->modelPath = modelPath;
+} 
+
+std::vector<CroppedTarget> Saliency::salience(cv::Mat image) { 
```

Comment:
This 80 line function should be broken up. I like that you are returning an object of our own type rather than these random IValue lists. I know you put a lot of effort into debugging Torch issues. It would be nice if you write down some of that knowledge in comments here. The code is quite terse and hard to tell what each section's intent is.

Reply:
Thank you for all of your feedback! I broke up `Salience::salience` into two functions to help with readability and SRP. I also added some more descriptive comments explaining what the code was doing, so hopefully it is easier to follow now.